### PR TITLE
board reset init'ed from user pf

### PR DESCRIPTION
**Pull request by maxzhen**
With this change, the reset will be initiated from user pf when you do xbutil reset (same as xbutil reset -h now). Xbutil reset -x and -f are added as hidden options for performing OCL reset and hot reset through xclmgmt respectively. They requires sudo and does not involve xocl any more (meant for debug usage only).

### CODE REVIEW

**Comment by maxzhen** [comment]
ok to test

**Comment by maxzhen** [comment]
ok to test

**Comment by stsoe** [comment]
ok to test

File: src/runtime_src/driver/xclng/drm/xocl/subdev/mb_scheduler.c [hunk 1]
```diff
@@ -1904,44 +1904,47 @@ static int
...
-	atomic_set(&client->abort, 0);
-	atomic_set(&client->outstanding_execs, 0);
-	client->num_cus = 0;
 	mutex_lock(&xdev->ctx_list_lock);
-	client->xdev = xocl_get_xdev(pdev);
-	list_add_tail(&client->link, &xdev->ctx_list);
 
-	/* kds must be configured on first xdev context even if that context
-	 * does not trigger an xclbin download */
-	if (list_is_singular(&xdev->ctx_list))
```
Comment (stsoe) [inline src/runtime_src/driver/xclng/drm/xocl/subdev/mb_scheduler.c]: Are you sure it safe to remove the reset_exec here on first context?

**Pull request merged**